ksql-engine/src/main/java/io/confluent/ksql/schema/registry/SchemaRegistryUtil.java

```diff
@@ -37,35 +41,51 @@ public static void cleanUpInternalTopicAvroSchemas(
       final String applicationId,
       final SchemaRegistryClient schemaRegistryClient
   ) {
-    getInternalSubjectNameSet(applicationId, schemaRegistryClient)
-        .forEach(subject -> tryDeleteSubject(applicationId, schemaRegistryClient, subject));
+    getInternalSubjectNames(applicationId, schemaRegistryClient)
+        .forEach(subject -> tryDeleteInternalSubject(applicationId, schemaRegistryClient, subject));
   }
 
-  private static Stream<String> getInternalSubjectNameSet(
-      final String applicationId,
-      final SchemaRegistryClient schemaRegistryClient
-  ) {
-    try {
-
-      return schemaRegistryClient.getAllSubjects().stream()
-          .filter(subjectName -> subjectName.startsWith(applicationId))
-          .filter(subjectName ->
-              subjectName.endsWith(CHANGE_LOG_SUFFIX) || subjectName.endsWith(REPARTITION_SUFFIX));
+  public static Stream<String> getSubjectNames(final SchemaRegistryClient schemaRegistryClient) {
+    return getSubjectNames(
+        schemaRegistryClient,
+        () -> "Could not get subject names from schema registry.");
+  }
 
+  private static Stream<String> getSubjectNames(
+      final SchemaRegistryClient schemaRegistryClient, final Supplier<String> errorMsgSupplier) {
+    try {
+      return schemaRegistryClient.getAllSubjects().stream();
     } catch (final Exception e) {
-      // Do nothing! Schema registry clean up is best effort!
-      LOG.warn("Could not clean up the schema registry for query: " + applicationId, e);
+      LOG.warn(errorMsgSupplier.get(), e);
```

Comment:
Just wondering, why is a Supplier<> object needed for just an error message?

From what I understand, suppliers are useful when we want wrap some code and execute it with get() just in some special cases, but this is just a string. Is there any other benefit?

Reply:
Yeah, the supplier does seem over engineering. Not like the cost of the string concat is a concern.